### ISSUE TITLE
New sets of components

### ISSUE THREAD

**Issue by flaviendelangle** [mui/mui-x#13896]
This issue aims at gathering all the requested sets of components in a single place.
If you want us to develop a set of component that is not listed here, please open a standalone issue, we will add it to the list below :+1: 

## Implemented

- Data Grid
- Date and Time Pickers
- Charts
- Tree View

## Requested 

- #510
- #511
- #513
- #514
- #1407
- #2901
- #3601
- #6152
- #12059
- #12364
- #13196
- #13232
- #13825
- #13883

**Comment by TheOneTheOnlyJJ** [comment]
I'm curious as to what is the status regarding most of these? Will their development be interleaved with all the upcoming Chart types? Is there any estimated time frame for the start of development for any of these?

**Comment by flaviendelangle** [comment]
Hi,

> Will their development be interleaved with all the upcoming Chart types? 

The charts now have a dedicated team which will be responsible for all the upcoming Chart types. Right now, we intend to continue giving a lot of bandwidth to this team, the charts are a project that has a lot of value for us and none of the components listed above will be realized at the expense of the charts. 

> Is there any estimated time frame for the start of development for any of these?

This list does not mean that we are interested in developing all of those, I just listed all the issues of people asking for new sets of components.
Among those components, we discussed a few internally and it is very likely that we are going to start one or two in the coming years (to best candidates are #2901 and #1407 but nothing written in stone for now).
The pace of development in the coming years will of course depend on our financial resources (to oversimplify more licenses sold => more developers => more components).

We may group the issues listed above between those we are sure to do in the coming years and those we are waiting to see the traction. But this has not been discussed internally yet :+1:

**Comment by TheOneTheOnlyJJ** [comment]
Isn't #514 an easier task compared to the rest of these thanks to the newly added [Rich Tree View Checkbox selection](https://mui.com/x/react-tree-view/rich-tree-view/selection/#checkbox-selection) customization? What's left is to wire such a Tree View to an input and keep the states in sync, right?

Also, I'd like to explicitly point out (if not discussed already) that #1407 would make sense to be developed along the Gantt Chart (#8732), which I can see on the [Public Roadmap](https://github.com/mui/mui-x/projects/1), as they go hand-in-hand and one's needs complement the other.

**Comment by flaviendelangle** [comment]
> Isn't https://github.com/mui/mui-x/issues/514 an easier task compared to the rest of these thanks to the newly added [Rich Tree View Checkbox selection](https://mui.com/x/react-tree-view/rich-tree-view/selection/#checkbox-selection) customization? What's left is to wire such a Tree View to an input and keep the states in sync, right?

I did not dive deep into this one honestly.
Looking at it quickly, it looks like something that should be achievable purely in userland if the Tree View has a good enough customization API.
Maybe we could provide a recipe for it and see if there is enough depth to add it on our side (maybe in the Tree View package).

**Comment by TheOneTheOnlyJJ** [comment]
> I did not dive deep into this one honestly. Looking at it quickly, it looks like something that should be achievable purely in userland if the Tree View has a good enough customization API. Maybe we could provide a recipe for it and see if there is enough depth to add it on our side (maybe in the Tree View package).

Yes, I agree that a recipe for it that could be copied & pasted into people's projects would be ideal if only state syncing is required. Bringing this to reality could only require a documentation update. If the team has the time, this could definitely be discussed and quickly looked at in a meeting.

> We won't start both the Gantt Chart and the Scheduler in parallel because it would increase the scope of the topic and make it harder to get something usable. But I totally agree that it should be easy to pass the same data to a Gant Chart and to a Scheduler to have two visualization of the same thing.

I was thinking the actual temporal content of the Scheduler could be directly represented in an embedded Gantt chart. Having seen this reply, I believe I was very wrong. Maybe my mental image of a Scheduler was very coupled with the Gantt chart and did not consider many alternative Scheduler features. Anyways, this is not the right issue for this topic.

> The Public Roadmap is not super up to date, I will add some components their later on 👍

Thanks for the clear updates you give every time!

**Comment by oliviertassinari** [comment]
It's worth noting that we try to rely as much as possible on the community to fill those gaps, e.g. for Material UI https://mui.com/material-ui/discover-more/related-projects/ lists all the extensions that we are aware of.

Now, if a component is highly requested, the community is not rising to fill the gap, and we can find a way to financially sustain the effort, you can expect us to go after the problem (there we find purpose: https://www.youtube.com/watch?v=lXLBTBBil2U&t=1420s).
